compact/test/Compiler.test.ts

```diff
@@ -0,0 +1,686 @@
+      );
+    });
+
+    it('should handle empty flags', async () => {
+      mockExec.mockResolvedValue({
+        stdout: 'Compilation successful',
+        stderr: '',
+      });
+
+      const result = await service.compileFile('MyToken.compact', '');
+
+      expect(result).toEqual({ stdout: 'Compilation successful', stderr: '' });
+      expect(mockExec).toHaveBeenCalledWith(
+        'compact compile "src/MyToken.compact" "artifacts/MyToken"',
+      );
+    });
+
+    it('should throw CompilationError when compilation fails', async () => {
+      mockExec.mockRejectedValue(new Error('Syntax error on line 10'));
+
+      await expect(
+        service.compileFile('MyToken.compact', '--skip-zk'),
+      ).rejects.toThrow(CompilationError);
+    });
+
+    it('should include file path in CompilationError', async () => {
+      mockExec.mockRejectedValue(new Error('Syntax error'));
+
+      try {
+        await service.compileFile('MyToken.compact', '--skip-zk');
+      } catch (error) {
+        expect(error).toBeInstanceOf(CompilationError);
+        expect((error as CompilationError).file).toBe('MyToken.compact');
+      }
+    });
+  });
+});
+
+describe('UIService', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.spyOn(console, 'log').mockImplementation(() => {});
+  });
+
+  describe('printOutput', () => {
+    it('should format output with indentation', () => {
+      const mockColorFn = vi.fn((text: string) => `colored(${text})`);
+
+      UIService.printOutput('line 1\nline 2\n\nline 3', mockColorFn);
+
+      expect(mockColorFn).toHaveBeenCalledWith(
+        '    line 1\n    line 2\n    line 3',
+      );
+      expect(console.log).toHaveBeenCalledWith(
+        'colored(    line 1\n    line 2\n    line 3)',
+      );
+    });
+
+    it('should handle empty output', () => {
+      const mockColorFn = vi.fn((text: string) => `colored(${text})`);
+
+      UIService.printOutput('', mockColorFn);
+
+      expect(mockColorFn).toHaveBeenCalledWith('');
+      expect(console.log).toHaveBeenCalledWith('colored()');
+    });
+  });
+
+  describe('displayEnvInfo', () => {
+    it('should display all environment information', () => {
+      UIService.displayEnvInfo(
+        'compact 0.1.0',
+        'Compactc 0.24.0',
+        'security',
+        '0.24.0',
+      );
+
+      // Test passes if no errors are thrown
+      expect(true).toBe(true);
+    });
+  });
+
+  describe('showCompilationStart', () => {
+    it('should show file count without target directory', () => {
+      UIService.showCompilationStart(5);
+
+      expect(true).toBe(true);
+    });
+
+    it('should show file count with target directory', () => {
+      UIService.showCompilationStart(3, 'security');
+
+      expect(true).toBe(true);
+    });
+  });
+
+  describe('showNoFiles', () => {
+    it('should show no files message', () => {
+      UIService.showNoFiles('security');
+
+      expect(true).toBe(true);
```

Comment:
I think instead of using expect(true).toBe(true) we should use the following pattern which is more readable.
```suggestion
       expect(() => UIService.displayEnvInfo(
        'compact 0.1.0',
        'Compactc 0.24.0',
        'security',
        '0.24.0',
      )).not.toThrow();
```